src/docs/Foundations/Typography/HeadingStyles/code.mdx

```diff
@@ -3,81 +3,124 @@ title: 'Code'
+
+<Highlight theme="warning" title="Keep in mind">
+
+You need to apply **at least two classes**, the block name and a size.
+
+</Highlight>
 
-```scss
-@import 'settings-tools/_all-settings';
 
-@include import-font-families();
 
-@import 'typography/_t.all-base-typography';
+```html
+<p class="mt-heading mt-heading--m">Lorem ipsum...</p>
 ```
 
-## Heading classes
+## Variations
 
-> Default lines-heights for the headings familly are set to `M`, witch correspond to a ratio close to 1.35.
-> Each heading atom comes with **a modifier to set the font-weight to 'semi bold'** and **a modifier to use an alternative `S` line-height**.
-> Headings atoms sizes change at the 'M' viewport, witch correspond to tablet sizes.
+### Sizes
 
-#### Import Headings atoms
+There is 3 Headings sizes available. Each heading size rescale it's font-size at the **M** and **L** viewports.
+Use the size modifiers in combination to the `mt-heading` class :
 
-```scss
-@import 'settings-tools/_all-settings';
+- **Small** : `mt-heading mt-heading--s`
+- **Medium** : `mt-heading mt-heading--m`
+- **Large** : `mt-heading mt-heading--l`
 
-@include import-font-families();
+<Preview path="typographyHeading-sizes" />
 
-@import 'typography/_t.headings';
-```
+| Name          | < screen-m (680px)   | >= screen-m (680px)  | screen-l (1024px)    |
+| ------------- | -------------------- | -------------------- | -------------------- |
+| **Heading S** | 07 / 23px (1.4375mu) | 08 / 28px (1.75mu)   | 09 / 34px (2.125mu)  |
+| **Heading M** | 08 / 28px (1.75mu)   | 09 / 34px (2.125mu)  | 10 / 41px (2.4625mu) |
+| **Heading L** | 09 / 34px (2.125mu)  | 10 / 41px (2.4625mu) | 11 / 49px (3.0625mu) |
 
-#### usage
 
-```html
-<p class="mt-heading-m">Lorem ipsum...</p>
-<p class="mt-heading-m mt-heading-m--semi-bold">Lorem ipsum...</p>
-<p class="mt-heading-m mt-heading-m--line-height-s">Lorem ipsum...</p>
-<p class="mt-heading-m mt-heading-m--line-height-s mt-heading-m--semi-bold">
-  Lorem ipsum...
-</p>
-```
+### Font-weights
+
+By default, the font-weight of all headings are set to `regular`.
+A semi-bold variation is available by applying the `mt-heading--semi-bold` modifier.
+
+<Preview path="typographyHeading-weight" />
+
+
+### Colors
+
+There is 2 text colors available for the Headings :
+
+- **Default** : the current color inherited from the context
+- **Lightest** : `mt-heading--lightest` (`$color-text-lightest` / `#fff` ) for use within a dark backgrounds.
+
+<Highlight theme="warning" title="Keep in mind">
+
+**The headings color should be the darkest color** (`$color-text-darkest` / `#000`) in most cases.
+We do not define the default color for you to be able to override it in very specific edge cases.
+
+</Highlight>
+
+<Preview path="typographyHeading-colors" />
+
+### Underlines
+
+Add underlines by applying the `mt-heading--underline` modifier.
+
+There is 4 underline color options for you to use, 2 for use within light backgrounds and 3 for use withing dark backgrounds :
```

Comment:
maybe something like: 
`There is a total of four underline color options for you to use. Two to use within light backgrounds and three to use within dark backgrounds:`